pkg/workloadmeta/collectors/internal/process/process.go

```diff
@@ -0,0 +1,145 @@
+		processData:     processData,
+		collectionClock: clock.New(),
+		pidToCid:        make(map[int]string),
+	}
+	return c
+}
+
+// Compile time check to ensure that `collector` implements `workloadmeta.Collector`.
+var _ workloadmeta.Collector = (*collector)(nil)
+
+type collector struct {
+	ddConfig config.ConfigReader
+
+	processData *checks.ProcessData
+
+	wlmExtractor *workloadmetaExtractor.WorkloadMetaExtractor
+	grpcServer   *workloadmetaExtractor.GRPCServer
+
+	pidToCid map[int]string
+
+	collectionClock clock.Clock
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Store) error {
+	if enabled, err := Enabled(c.ddConfig); !enabled {
+		return err
+	}
+
+	err := c.grpcServer.Start()
+	if err != nil {
+		return err
+	}
+
+	collectionTicker := c.collectionClock.Ticker(
+		c.ddConfig.GetDuration("workloadmeta.local_process_collector.collection_interval"),
+	)
+
+	filter := workloadmeta.NewFilter([]workloadmeta.Kind{workloadmeta.KindContainer}, workloadmeta.SourceAll, workloadmeta.EventTypeAll)
+	containerEvt := store.Subscribe(collectorId, workloadmeta.NormalPriority, filter)
+
+	go c.run(ctx, store, containerEvt, collectionTicker)
+
+	return nil
+}
+
+func (c *collector) run(ctx context.Context, store workloadmeta.Store, containerEvt chan workloadmeta.EventBundle, collectionTicker *clock.Ticker) {
+	defer c.grpcServer.Stop()
+	defer store.Unsubscribe(containerEvt)
+	defer collectionTicker.Stop()
+
+	for {
+		select {
+		case evt := <-containerEvt:
+			c.handleContainerEvent(evt)
+		case <-collectionTicker.C:
+			err := c.processData.Fetch()
+			if err != nil {
+				_ = log.Error("Error fetching process data:", err)
+			}
+		case <-ctx.Done():
+			log.Infof("The %s collector has stopped", collectorId)
+			return
+		}
+	}
+}
+
+// Pull is unused at the moment used due to the short frequency in which it is called.
+// In the future, we should use it to locally in workload-meta.
+func (c *collector) Pull(_ context.Context) error {
+	return nil
+}
+
+func (c *collector) handleContainerEvent(evt workloadmeta.EventBundle) {
+	defer close(evt.Ch)
+
+	for _, evt := range evt.Events {
+		ent := evt.Entity.(*workloadmeta.Container)
+		switch evt.Type {
+		case workloadmeta.EventTypeSet:
+			// Should be safe, even on windows because PID 0 is the idle process and therefore must always belong to the host
+			if ent.PID != 0 {
+				c.pidToCid[ent.PID] = ent.ID
+			}
+		case workloadmeta.EventTypeUnset:
+			delete(c.pidToCid, ent.PID)
+		}
+	}
+
+	c.wlmExtractor.SetLastPidToCid(c.pidToCid)
+}
+
+func Enabled(cfg config.ConfigReader) (bool, error) {
+	if cfg.GetBool("process_config.process_collection.enabled") {
+		return false, dderrors.NewDisabled(collectorId, "the process check is enabled")
+	}
+
+	if !cfg.GetBool("workloadmeta.remote_process_collector.enabled") {
+		return false, dderrors.NewDisabled(collectorId, "the process collector is disabled")
+	}
+	return true, nil
+}
```

Comment:
I think we should also use flavor to only enable this in the process agent.

Also, is it necessary to return errors ? Perhaps logs could be enough.

Reply:
> I think we should also use flavor to enable this only in the process agent.
That's fair, although I have a feeling that we misused flavor a bit, since it's meant to represent the environment the agent is running in, not the current binary. I tried to avoid it here so that it can be easily changed in the future. I am happy to change this if you feel strongly about it however!

> Also, is it necessary to return errors ? Perhaps logs could be enough.
It seems that this is the standard for workloadmeta collectors, no? For example it is used [here](https://github.com/DataDog/datadog-agent/blob/5eb542ef374c942a1ca9c4fff9f2511b89a5145e/pkg/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go#L43-L45) in the ECS fargate collector

Reply:
> I think we should also use flavor to enable this only in the process agent.

Added!

> Also, is it necessary to return errors ? Perhaps logs could be enough.

Discussed offline, other collectors do this, so it should be okay here.